File: 162717909/credit/credit.c
#include <cs50.h>
#include <stdio.h>
int main()
{
    long cardnum = 0;
    do
    {
		cardnum = get_long("Number: ");
	}
	while (cardnum <= 0);
	// printf("%ld\n",cardnum);
	int sum = 0;
	int flag = 0;
	int num = 0;
	while (cardnum > 0)
	{
		if (cardnum < 100 && cardnum > 10)
		{
			flag = cardnum % 10;
		}
		if (cardnum < 10)
		{
			flag = flag + cardnum % 10 * 10;
		}
		sum = sum + cardnum % 10;
		cardnum = cardnum / 10;
		num++;
		// printf("cardnum=%ld,flag=%d\n",cardnum,flag);
		if (cardnum > 0)
		{
				if (cardnum < 100 && cardnum > 10)
			{
				flag = cardnum % 10;
			}
			if (cardnum < 10)
			{
				flag = flag + cardnum % 10 * 10;
			}
			int split = 0;
			split = cardnum % 10;
			split = split * 2;
			while (split > 0)
			{
				sum = sum + split % 10;
				split = split / 10;
			}
			cardnum = cardnum / 10;
			num++;
			// printf("cardnum=%ld,flag=%d\n",cardnum,flag);
		}
	}
	if (sum % 10 == 0 && (13 <= num && num <= 16))
	{
		if ((flag == 34 || flag == 37) && num == 15)
		{
			printf("AMEX\n");
		}
		else if (40 <= flag && flag < 50 && (num == 13 || num == 16))
		{
			printf("VISA\n");
		}
		else if (50 <= flag && flag < 56 && num == 16)
		{
			printf("MASTERCARD\n");
		}
		else
		{
			printf("INVALID\n");
		}
	}
	else
	{
		printf("INVALID\n");
	}
	//printf("sum = %d, num = %d, flag=%d", sum, num, flag);
	return 0;
 }

File: 162717909/sentimental-credit/credit.py
from cs50 import get_int
card = get_int("Number: ")

sum = 0
head = str(card)
head = head[0:2]
length = len(str(card))

while card > 0:
    a = card % 10
    sum += a
    card = card // 10
    b = card % 10
    b *= 2
    card = card // 10
    while b > 0:
        sum = sum + b % 10
        b = b // 10

# print(sum)
# print(sum%10)
# print(head)
# print(card)
#  card值已改变，前面要用一个值储存card值
if length in [13, 15, 16] and sum % 10 == 0:
    if head == '34' or head == '37':
        print("AMEX")
    if head[0] == '4':
        print("VISA")
    if head == '51' or head == '52' or head == '53' or head == '54' or head == '55':
        print("MASTERCARD")
    else:
        print("INVALID")
else:
    print("INVALID")
